Showcase-iOS/About/View/AboutViewController.swift

```diff
@@ -32,44 +24,49 @@ class AboutViewController: UIViewController {
         aboutPresenter?.retrieveSocialMediaLinks()
     }
 
-    // MARK: Opertaions
+    // MARK: Opertaion(s)
 
     func setupInjectables() {
         let aboutPresenter = dependencyContainer.resolve(AboutPresentable.self) as? AboutPresenter
         let firebaseStorage = dependencyContainer.resolve(FIRStoring.self)
         self.aboutPresenter = aboutPresenter
         aboutPresenter?.aboutView = self
+        aboutPresenter?.analyticManager = dependencyContainer.resolve(AnalyticsManager.self) as? AnalyticManagerImplementation
         self.firebaseStorage = firebaseStorage
     }
 
     // MARK: @IBActions
 
     @IBAction func websiteTapped(_ sender: Any) {
+        aboutPresenter?.trackSocialMediaButtonTap(with: "website")
         if let websiteUrl = aboutViewModel?.website {
             UIApplication.shared.open(websiteUrl, options: [:], completionHandler: nil)
         }
     }
 
     @IBAction func twitterTapped(_ sender: Any) {
+        aboutPresenter?.trackSocialMediaButtonTap(with: "twitter")
         if let twitterUrl = aboutViewModel?.twitter {
             UIApplication.shared.open(twitterUrl, options: [:], completionHandler: nil)
         }
     }
 
     @IBAction func facebookTapped(_ sender: Any) {
+        aboutPresenter?.trackSocialMediaButtonTap(with: "facebook")
```

Comment:
this code also seems duplicated, Please can you write a function that takes in the presenter, the enum and the url string and does this UIApplication.shared.open functionality

Reply:
done